src/Components/Aspire.Microsoft.EntityFrameworkCore.Cosmos/AspireAzureEFCoreCosmosExtensions.cs

```diff
@@ -23,6 +23,47 @@ public static class AspireAzureEFCoreCosmosExtensions
     private const string DefaultConfigSectionName = "Aspire:Microsoft:EntityFrameworkCore:Cosmos";
     private const DynamicallyAccessedMemberTypes RequiredByEF = DynamicallyAccessedMemberTypes.PublicConstructors | DynamicallyAccessedMemberTypes.NonPublicConstructors | DynamicallyAccessedMemberTypes.PublicProperties;
 
+    /// <summary>
+    /// Registers the given <see cref="DbContext" /> as a service in the services provided by the <paramref name="builder"/>.
+    /// Derives the name of the database from the connection string.
+    /// </summary>
+    /// <typeparam name="TContext">The <see cref="DbContext" /> that needs to be registered.</typeparam>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="connectionName">A name used to retrieve the connection string from the ConnectionStrings configuration section.</param>
+    /// <param name="configureSettings">An optional delegate that can be used for customizing settings. It's invoked after the settings are read from the configuration.</param>
+    /// <param name="configureDbContextOptions">An optional delegate to configure the <see cref="DbContextOptions"/> for the context.</param>
+    /// <exception cref="ArgumentNullException">Thrown if mandatory <paramref name="builder"/> is null.</exception>
+    /// <exception cref="InvalidOperationException">Thrown when mandatory <see cref="EntityFrameworkCoreCosmosSettings.ConnectionString"/> is not provided.</exception>
+    public static void AddCosmosDbContext<[DynamicallyAccessedMembers(RequiredByEF)] TContext>(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<EntityFrameworkCoreCosmosSettings>? configureSettings = null,
+        Action<DbContextOptionsBuilder>? configureDbContextOptions = null) where TContext : DbContext
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+        ArgumentException.ThrowIfNullOrEmpty(connectionName);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
```

Comment:
What happens when the connection string isn't provided? Typically we still try to add the services and we expect the app code in `configureSettings` to provide the connection information - instead of the connection string.

Reply:
This method should throw if the connection string isn't provided.
Advanced usage where the connection string is set later would have to use `AddDbContext` with [UseCosmos](https://learn.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.cosmosdbcontextoptionsextensions.usecosmos?view=efcore-9.0#microsoft-entityframeworkcore-cosmosdbcontextoptionsextensions-usecosmos(microsoft-entityframeworkcore-dbcontextoptionsbuilder-system-action((microsoft-entityframeworkcore-infrastructure-cosmosdbcontextoptionsbuilder)))) and `EnrichCosmosDbContext`

Reply:
> This method should throw if the connection string isn't provided.

We also support setting the connection properties via the `Action<EntityFrameworkCoreCosmosSettings>? configureSettings` parameter for the cases where apps need to retrieve this information from somewhere other than the connection string section.

Reply:
> We also support setting the connection properties via the Action<EntityFrameworkCoreCosmosSettings>? configureSettings parameter for the cases where apps need to retrieve this information from somewhere other than the connection string section.

If both are specified which one should take precedence?

Reply:
> If both are specified which one should take precedence?

I've implemented it such that the name defined in `configureSettings` takes precedence.

Reply:
> I've implemented it such that the name defined in configureSettings takes precedence.

👍 That follows the same pattern in all our client integrations.

* Settings object is bound to the configuration
* Values coming from the connection string are applied to the settings object
* configureSettings is called, so code and inspect and modify it.